plugins/idea/pluginBase/src/main/java/com/imaginea/kodebeagle/base/util/SearchUtils.java

```diff
@@ -70,55 +78,48 @@ public final int getResultCount() {
         return resultCount;
     }
 
+
     public final void fetchContentsAndUpdateMap(final List<String> fileNames) {
-        String esFileQueryJson = jsonUtils.getJsonForFileContent(fileNames);
-        String esFileResultJson;
-        esFileResultJson = getESResultJson(esFileQueryJson,
-                windowObjects.getEsURL() + SOURCEFILE_SEARCH);
-        JsonObject jsonElements = getJsonElements(esFileResultJson);
-        if (jsonElements != null) {
-            JsonArray hitsArray = getJsonHitsArray(jsonElements);
-            for (JsonElement hits : hitsArray) {
-                JsonObject hitObject = hits.getAsJsonObject();
-                JsonObject sourceObject = hitObject.getAsJsonObject(SOURCE);
-                //Replacing \r as it's treated as bad end of line character
-                String fileContent = sourceObject.getAsJsonPrimitive(FILE_CONTENT).
-                        getAsString();
-                String fileName = sourceObject.getAsJsonPrimitive(FILE_NAME).getAsString();
-                if (fileName != null && !fileName.isEmpty()
-                        && fileContent != null && !fileContent.isEmpty()) {
+
+            for (String fileName : fileNames) {
+                if (!fileName.isEmpty()) {
+
+                    String  kbFileResultJson = getResponseJson(windowObjects.getKbAPIURL() + SOURCEFILE_SEARCH, fileName, true);
+
+
+
                     windowObjects.getFileNameContentsMap().put(fileName,
-                            fileContent.replaceAll("\r", ""));
+                            kbFileResultJson);
                 }
             }
-        }
     }
 
     public final String getContentsForFile(final String fileName) {
         Map<String, String> fileNameContentsMap =
                 windowObjects.getFileNameContentsMap();
 
         if (!fileNameContentsMap.containsKey(fileName)) {
+
             fetchContentsAndUpdateMap(Arrays.asList(fileName));
+
         }
         String fileContent = fileNameContentsMap.get(fileName);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "fileContent". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)